include/fluent-bit/flb_output.h

```diff
@@ -526,6 +565,125 @@ static FLB_INLINE void output_pre_cb_flush(void)
+    if (flb_output_is_threaded(o_ins) == FLB_TRUE) {
+        th_ins = flb_output_thread_instance_get();
+        pthread_mutex_lock(&th_ins->timer_mutex);
+        mk_list_del(&timer_coro->_head);
+        mk_list_add(&timer_coro->_head, &th_ins->timer_coro_list_destroy);
+        pthread_mutex_unlock(&th_ins->timer_mutex);
+    }
+    else {
+        mk_list_del(&timer_coro->_head);
+        mk_list_add(&timer_coro->_head, &o_ins->timer_coro_list_destroy);
+    }
+
+    /* timer coro is complete; yield back to caller/control code */
+    flb_coro_yield(coro, FLB_TRUE);
+}
+
+/* 
+ * If the output uses scheduled timers with coroutines,
+ * this function is used as the callback for flb_sched_timer_cb_create
+ */
+static FLB_INLINE
+void flb_output_coro_timer_cb(struct flb_config *config, void *data)
+{
+    size_t stack_size;
+    struct flb_coro *coro;
+    struct flb_output_timer_coro *timer_coro;
+    struct flb_out_thread_instance *th_ins;
+    struct flb_output_coro_timer_data *ctx = (struct flb_output_coro_timer_data *) data;
+    struct flb_out_timer_coro_params *params;
+    struct flb_output_instance *o_ins;
+
+    /* Custom output coroutine info */
+    timer_coro = (struct flb_output_timer_coro *) flb_calloc(1, sizeof(struct flb_output_timer_coro));
+    if (!timer_coro) {
+        flb_errno();
+        return;
+    }
+
+    /* Create a new co-routine */
+    coro = flb_coro_create(timer_coro);
+    if (!coro) {
+        flb_free(timer_coro);
+        return;
+    }
+
+    o_ins = ctx->ins;
+    timer_coro->o_ins  = o_ins;
+    timer_coro->config = config;
+    timer_coro->coro   = coro;
+
+    coro->callee = co_create(config->coro_stack_size,
+                             output_pre_timer_cb, &stack_size);
+
+    if (coro->callee == NULL) {
+        flb_coro_destroy(coro);
+        flb_free(timer_coro);
+        return;
+    }
+
+#ifdef FLB_HAVE_VALGRIND
+    coro->valgrind_stack_id = \
+        VALGRIND_STACK_REGISTER(coro->callee, ((char *) coro->callee) + stack_size);
+#endif
+
+    if (o_ins->is_threaded == FLB_TRUE) {
+        th_ins = flb_output_thread_instance_get();
+        pthread_mutex_lock(&th_ins->timer_mutex);
+        mk_list_add(&timer_coro->_head, &th_ins->timer_coro_list);
+        pthread_mutex_unlock(&th_ins->timer_mutex);
+    }
+    else {
+        mk_list_add(&timer_coro->_head, &o_ins->timer_coro_list);
+    }
+
+    params = (struct flb_out_timer_coro_params *) FLB_TLS_GET(timer_coro_params);
+    if (!params) {
+        params = (struct flb_out_timer_coro_params *) flb_calloc(1, sizeof(struct flb_out_flush_params));
+        if (!params) {
+            flb_errno();
+            return;
+        }
+    }
+
+    /* Callback parameters in order */
+    params->output_timer = timer_coro;
+    params->timer_data = ctx;
+    params->config      = config;
+    params->coro        = coro;
+
+    FLB_TLS_SET(timer_coro_params, params);
+    coro->caller = co_active();
+    flb_coro_resume(coro);
```

Comment:
It appears that this will begin the callback from the start. From this point on the network events will take over control

Reply:
Yea that's the intention